### DISCUSSION TITLE
How can I version the chunk names in remix?

### DISCUSSION THREAD

**Discussion by nabeelpkl** [remix-run/remix#8274] [Q&A]
How can I version the chunk names in remix, so that its different for every release? During every release I want the cache to get revalidated automatically. Is something like this possible in remix? 
[webpack solution](https://stackoverflow.com/a/34090319)

**Comment by alexanderson1993** [comment]
Remix already does this for any browser assets, including your JavaScript files and CSS files. You can safely mark those files as "immutable" with cache headers, since their filenames will change every release. 


Here's demonstration from one of the Remix Examples that uses Express (the Remix App Server does this for you automatically): https://github.com/remix-run/examples/blob/main/dataloader/server/index.ts#L20

**Reply by nabeelpkl** [reply]
@alexanderson1993 but one doubt here, even after remix doing this why is the chunk names still same? when does this gets updated ? ideally I want this to change after deployment so that further request should not fetch it from cache. can you please help me understand this

**Reply by aarjithn** [reply]
@nabeelpkl Usually chunk names are hashed to content so it would only change if underlying file content changes. This means if you just change a file for release, only that chunk will be renamed, and others retain the cached behaviour.

**Reply by sergiodxa** [reply]
Note that if chunk A imports chunk B and chunk B changes, both chunks will change since the chunk A needs to reference the new hash one the chunk B, so a single file change could cause multiple chunks to change the hash, although it shouldn't happen in every chunk.

**Reply by nabeelpkl** [reply]
Thanks @aarjithn @sergiodxa. I was asking this because I am seeing my web app crashes after any major changes are deployed. I have enabled browser caching for 30 mints and cdn caching for 8hrs. App doesnt crash when I disable the caching. Any suggestion to resolve this?

**Reply by sergiodxa** [reply]
Are you caching the HTML?

**Reply by nabeelpkl** [reply]
Not yet, but planning to do that as in few cases page is taking time to load.

**Reply by aarjithn** [reply]
@nabeelpkl If your app crashes, that means the file that is being requested is not present. Check which js file is getting 404, and if that is actually part of the new build. If not, then you are indeed caching the html.

**Reply by nabeelpkl** [reply]
@sergiodxa @aarjithn I am caching the `build` folder. The caching is handled from cloudfront which overrides the app caching rules. will this be a reason for the issue?
